packages/node_modules/@ciscospark/react-component-adaptive-card/src/index.js

```diff
@@ -17,11 +21,53 @@ class AdaptiveCard extends React.Component {
   componentDidMount() {
     if (Object.prototype.hasOwnProperty.call(this, 'nodeElement') && Object.prototype.hasOwnProperty.call(this.nodeElement, 'current')) {
       this.nodeElement.current.appendChild(
-        getAdaptiveCard(this.props.cards, this.props.displayName, this.props.sparkInstance, this.addChildNode)
+        getAdaptiveCard(this.state.cards, this.props.displayName, this.props.sparkInstance, this.addChildNode)
       );
     }
   }
 
+  static getDerivedStateFromProps(nextProps) {
+    if (nextProps.verb === API_ACTIVITY_VERB.SHARE) {
+      try {
+        const decryptedURLs = nextProps.items
+          .filter((file) => file.type === CARD_CONTAINS_IMAGE)
+          .map((file) => {
+            const thumbnail = file.mimeType === 'image/gif' ? nextProps.share.getIn(['files', file.url]) : nextProps.share.getIn(['files', file.image.url]);
+
+            if (thumbnail) {
+              const objectUrl = thumbnail.get('objectUrl');
+
+              return objectUrl;
+            }
+
+            return undefined;
+          });
+        const cards = JSON.parse(nextProps.cards[0]);
+        const undefinedUrls = decryptedURLs.filter((file) => file === undefined);
```

Comment:
What's the purpose of returning `undefined` earlier if they're going to be filtered out? wouldn't it be better to just not add them (by not returning a few lines earlier) to begin with?

Reply:
Initially, there is a chance to get the URL Value as undefined instead of the index. So filtering the undefinedUrls and  if there is no content in thumbnail also arrow function will expect a return value